PSReadLine/StringBuilderLinewiseExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.PowerShell
+{
+    public class Range

```

Comment:
This goes hand in hand with why `StringBuilderLinewiseExtensions` should be made public. In essence, this does not _need_ to be public. It’s just that I am using TDD to implement my developments and I do not see an elegant way to test `StringBuilderLinewiseExtensions` without it being public.

That said, the public / external _behavior_ of the code is ultimately unit-tested as well.

Would you be OK if I added some `#if DEBUG` conditionals so that this code is tested during debug builds only ? Keeping in mind that the purpose of this code is to have a few public facing API that are always tested anyway (in my case, by new tests in the `YankPasteTest.VI.cs` file).

Another solution would be to use a dummy .snk strong-key to sign both assemblies and make use of the `InternalsVisibleTo` attributes.

As a third solution, I would make those classes non-public as appropriate and remove the corresponding unit-tests.

Can you weigh in on this issue ? Alternatively, I can ask a question as a separate issue to have a broader consensus involving more developers you may trust.

Reply:
I'll let @daxian-dbw chime in here. In the past we used `InternalsVisibleTo` but that may be less desirable now.

Another alternative is for a "test only" build where the unit tests are included in the same binary.

Private reflection is yet another (probably wrong) option.

Reply:
I like the idea to include the tests in the same assembly with conditional compilation.